extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/query/AttributeSubsettingIterator.java

```diff
@@ -0,0 +1,206 @@
+		final List<byte[]> valsToKeep = new ArrayList<>();
+		int totalSize = 0;
+		final List<Integer> originalPositions = BitmaskUtils.getFieldPositions(originalBitmask);
+		// convert list to set for quick contains()
+		final Set<Integer> newPositions = new HashSet<Integer>(
+				BitmaskUtils.getFieldPositions(newBitmask));
+		if (originalPositions.size() > 1) {
+			for (final Integer originalPosition : originalPositions) {
+				final int len = originalBytes.getInt();
+				final byte[] val = new byte[len];
+				originalBytes.get(val);
+				if (newPositions.contains(originalPosition)) {
+					valsToKeep.add(val);
+					totalSize += len;
+				}
+			}
+		}
+		else if (!newPositions.isEmpty()) {
+			// this shouldn't happen because we should already catch the case
+			// where the bitmask is unchanged
+			return original;
+		}
+		else {
+			// and this shouldn't happen because we should already catch the
+			// case where the resultant bitmask is empty
+			return null;
+		}
+		if (valsToKeep.size() == 1) {
+			final ByteBuffer retVal = ByteBuffer.allocate(totalSize);
+			retVal.put(valsToKeep.get(0));
+			return new Value(
+					retVal.array());
+		}
+		final ByteBuffer retVal = ByteBuffer.allocate((valsToKeep.size() * 4) + totalSize);
+		for (final byte[] val : valsToKeep) {
+			retVal.putInt(val.length);
+			retVal.put(val);
+		}
+		return new Value(
+				retVal.array());
+	}
+
+	@Override
+	public void init(
+			final SortedKeyValueIterator<Key, Value> source,
+			final Map<String, String> options,
+			final IteratorEnvironment env )
+			throws IOException {
+		super.init(
+				source,
+				options,
+				env);
+		// get fieldIds and associated adapter
+		final String bitmaskStr = options.get(FIELD_SUBSET_BITMASK);
+		fieldSubsetBitmask = ByteArrayUtils.byteArrayFromString(bitmaskStr);
+	}
+
+	@Override
+	public boolean validateOptions(
+			final Map<String, String> options ) {
+		if ((!super.validateOptions(options)) || (options == null)) {
+			return false;
+		}
+		final boolean hasFieldsBitmask = options.containsKey(FIELD_SUBSET_BITMASK);
+		if (!hasFieldsBitmask) {
+			// all are required
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * 
+	 * @return an {@link IteratorSetting} for this iterator
+	 */
+	public static IteratorSetting getIteratorSetting() {
+		return new IteratorSetting(
+				AttributeSubsettingIterator.ITERATOR_PRIORITY,
+				AttributeSubsettingIterator.ITERATOR_NAME,
+				AttributeSubsettingIterator.class);
+	}
+
+	/**
+	 * Sets the desired subset of fields to keep
+	 * 
+	 * @param setting
+	 *            the {@link IteratorSetting}
+	 * @param adapterAssociatedWithFieldIds
+	 *            the adapter associated with the given fieldIds
+	 * @param fieldIds
+	 *            the desired subset of fieldIds
+	 * @param numericDimensions
+	 *            the numeric dimension fields
+	 */
+	public static void setFieldIds(
+			final IteratorSetting setting,
+			final DataAdapter<?> adapterAssociatedWithFieldIds,
```

Comment:
Regarding the and operation on different length byte arrays, the operation produces the smallest length as the result - essentially if bytes are missing at the end they are assumed 0 or missing fields which should accurately reflect the behavior.

If someone were to add fields, correct, it would have to be at the end.

I don't think endian-ness comes into play here - its always bitwise operations on raw bytes without any translation to numbers.  As long as the bytes come across consistently in serialization/deserialization (which they will) after that point its just bitwise operations.